### ISSUE TITLE
Function parameter typed as never is still required

### ISSUE THREAD

**Issue by wcandillon** [microsoft/TypeScript#36895]
https://www.typescriptlang.org/play/?ssl=4&ssc=5&pln=4&pc=10#code/C4TwDgpgBAYg9nKBeKAKAhgJwOYC4oB2EAbhJgJTIB8UARggDYToEDcAUOwMZwEDOwKADME+eIhSpKSGsEwBXCBxFwprIA

It looks like the example below shouldn't throw an error since arg is of type never.
```ts
type Foo = (arg: never) => boolean;

const foo: Foo = () => true;
foo();
```

**Comment by wcandillon** [comment]
Fixed it by replacing never with void, thanks!
